Autopilot/Test/Inc/batterysensor_mock.hpp

```diff
@@ -0,0 +1,15 @@
+#ifndef BATTERYSENSOR_MOCK_HPP
+#define BATTERYSENSOR_MOCK_HPP
+
+#include <gmock/gmock.h>
+
+#include "batterysensor.hpp"
+
+class MockBatterySensor : public batterySensor
```

Comment:
This inherits from the class "batterySensor", which, to Lucy's point doesn't exist.

Reply:
Sorry; got blocked by work - my understanding so far was that these were to be mock implementations and didn't require a class to be written, although that seems to be incorrect lol. Do you want me to write a batterySensor class then or just remove the inheritment?

Reply:
I recommend google some examples of google mock so you can understand what's going on.

But long story short, your interface should be an abstract class. During test time, the mock class derives from it, and during real operation, the actual driver class derives from it. Lucy has a full working example in her PR as well, take a look.

Reply:
Alright, I think i've done it right in the most recent commit